brave/src/test/java/brave/secondary_sampling/TestSecondarySampler.java

```diff
@@ -56,6 +69,12 @@ public SecondarySampler forService(String serviceName) {
       Trigger trigger = getByService(state.samplingKey()).get(serviceName);
       if (trigger == null) trigger = allServices.get(state.samplingKey());
       if (trigger == null) return false;
+
+      // When in passive mode, we only evaluate when upstream did
+      if (trigger.mode == Trigger.Mode.PASSIVE) {
```

Comment:
@narayaruna this is the actual bug fix on RPS

Reply:
This looks good. 

I would like to provide some insight in to what I was originally thinking when we designed this.

I was thinking to support only PASSIVE sampling. This means, presence of a `sampling_key` indicate the upstream is sampled. Downstream services can only participate passively.

**If a service is participating only passively then how can we support the `ACTIVE` use case?**
The way I originally thought about this, to support this use case, it will be a new secondary sampling started in that service. 

To explain this with an example;
`gateway -> api -> auth -> cache  -> authdb`

To start active sampling for `auth`, a new secondary sampling trace will be started in the `auth` service and the `sampling_key` will be a new one applicable to `auth` and downstream.